File: Java/LeetCode/lc2500/Solution.java
package lc2500;

class Solution {
    public int deleteGreatestValue(int[][] grid) {
        int ret = 0;
        int rows = grid.length;

        for(int[] row : grid){
            Arrays.sort(row);
        }

        int col = grid[0].length;

        for(int i = 0; i < col; i++){
            int maxValueCol = 0;
            for(int j = 0; j < rows; j++){
                maxValueCol = Math.max(maxValueCol,grid[j][i]);
            }
            ret += maxValueCol;
        }

        return ret;
    }
}

File: TypeScript/LeetCode/lc2500/solution.ts
function deleteGreatestValue(grid: number[][]): number {
    var ret = 0;
    var rows = grid.length;
    for(var i = 0; i < rows; i++){
        grid[i].sort((a,b)=>a-b);
    }

    for(var col = 0;  col < grid[0].length; col++){
        var maxValueCol = 0;
        for(var row = 0; row < rows; row++){
            maxValueCol = Math.max(maxValueCol,grid[row][col]);
        }
        ret += maxValueCol;
    }

    return ret;
};